source/containerimage/pull.go

```diff
@@ -73,20 +88,41 @@ func (is *Source) ResolveImageConfig(ctx context.Context, ref string, opt llb.Re
 		dgst digest.Digest
 		dt   []byte
 	}
+	var typed *t
 	key := ref
 	if platform := opt.Platform; platform != nil {
 		key += platforms.Format(*platform)
 	}
+	var (
+		rm    source.ResolveMode
+		rslvr remotes.Resolver
+		err   error
+	)
+
+	switch is.Source {
```

Comment:
I'm bit confused about how this works. I would think you would need to pass a `oci-layout` ref and sessionid in the `ResolveImageConfigOpt` struct. Will look again if the test passes.

Reply:
Most of the changes here are just getting the resolver, so that it can be passed to the `imageutil.Config()`.

As far as the `sessionID`, I agree, but it isn't in `ResolveImageConfigOpt` (unlike the call to `Resolve()`, which includes `id source.Identifier`, which includes the `SessionID`.

I was not sure how to fix that, so I worked around it.